[gatsby-image] Bad schema typing for gatsby-images ?

Hi,

The TS defs for a fluid image is:

```tsx
export interface FluidObject {
  aspectRatio: number
  src: string
  srcSet: string
  sizes: string
  base64?: string
  tracedSVG?: string
  srcWebp?: string
  srcSetWebp?: string
  media?: string
}
```

Yet if we look at a graphql image fragment like this one, we can notice they are actually incompatible:

![image](https://user-images.githubusercontent.com/749374/63606583-b4480900-c5d0-11e9-9a36-54dc513158fb.png)


The GraphQL schema says that all those attributes can be undefined (which does not really make sense to me, is it possible to have an image without at least an src?)

The problem is that if you try to generate typedefs from generic queries, you'll end up with incompatible typings and will have to find workarounds (like using "any").

Otherwise you'll get typescript errors like this:

```
[0] src/page/home/queries/Images.tsx(54,3): error TS2322: Type 'Pick<ImageSharpFluid, "src" | "sizes" | "srcSet" | "aspectRatio" | "tracedSVG" | "srcWebp" | "srcSetWebp">[]' is not assignable to type 'Pick<FluidObject, "media" | "src" | "base64" | "sizes" | "srcSet" | "tracedSVG" | "srcWebp" | "srcSetWebp">[]'.
[0]   Type 'Pick<ImageSharpFluid, "src" | "sizes" | "srcSet" | "aspectRatio" | "tracedSVG" | "srcWebp" | "srcSetWebp">' is not assignable to type 'Pick<FluidObject, "media" | "src" | "base64" | "sizes" | "srcSet" | "tracedSVG" | "srcWebp" | "srcSetWebp">'.
```

I suggest to make those types compatible so that people using tools like graphql-codegen can easily use gatsby images

--- a/packages/gatsby-source-contentful/src/extend-node-type.js
+++ b/packages/gatsby-source-contentful/src/extend-node-type.js
@@ -6,6 +6,7 @@ const {
   GraphQLInt,
   GraphQLFloat,
   GraphQLJSON,
+  GraphQLNonNull,
 } = require(`gatsby/graphql`)
 const qs = require(`qs`)
 const base64Img = require(`base64-img`)
@@ -327,10 +328,10 @@ const fixedNodeType = ({ name, getTracedSVG }) => {
           resolve: getTracedSVG,
         },
         aspectRatio: { type: GraphQLFloat },
-        width: { type: GraphQLFloat },
-        height: { type: GraphQLFloat },
-        src: { type: GraphQLString },
-        srcSet: { type: GraphQLString },
+        width: { type: new GraphQLNonNull(GraphQLFloat) },
+        height: { type: new GraphQLNonNull(GraphQLFloat) },
+        src: { type: new GraphQLNonNull(GraphQLString) },
+        srcSet: { type: new GraphQLNonNull(GraphQLString) },
         srcWebp: {
           type: GraphQLString,
           resolve({ image, options, context }) {
@@ -421,9 +422,9 @@ const fluidNodeType = ({ name, getTracedSVG }) => {
           type: GraphQLString,
           resolve: getTracedSVG,
         },
-        aspectRatio: { type: GraphQLFloat },
-        src: { type: GraphQLString },
-        srcSet: { type: GraphQLString },
+        aspectRatio: { type: new GraphQLNonNull(GraphQLFloat) },
+        src: { type: new GraphQLNonNull(GraphQLString) },
+        srcSet: { type: new GraphQLNonNull(GraphQLString) },
         srcWebp: {
           type: GraphQLString,
           resolve({ image, options, context }) {
@@ -458,7 +459,7 @@ const fluidNodeType = ({ name, getTracedSVG }) => {
             return _.get(fluid, `srcSet`)
           },
         },
-        sizes: { type: GraphQLString },
+        sizes: { type: new GraphQLNonNull(GraphQLString) },
       },
     }),
     args: {

--- a/packages/gatsby-transformer-sharp/src/customize-schema.js
+++ b/packages/gatsby-transformer-sharp/src/customize-schema.js
@@ -75,9 +75,9 @@ const fixedNodeType = ({
         },
         aspectRatio: { type: GraphQLFloat },
         width: { type: new GraphQLNonNull(GraphQLFloat) },
-        height: { type: GraphQLNonNull(GraphQLFloat) },
-        src: { type: GraphQLNonNull(GraphQLString) },
-        srcSet: { type: GraphQLNonNull(GraphQLString) },
+        height: { type: new GraphQLNonNull(GraphQLFloat) },
+        src: { type: new GraphQLNonNull(GraphQLString) },
+        srcSet: { type: new GraphQLNonNull(GraphQLString) },
         srcWebp: {
           type: GraphQLString,
           resolve: ({ file, image, fieldArgs }) => {

--- a/packages/gatsby-transformer-sharp/src/customize-schema.js
+++ b/packages/gatsby-transformer-sharp/src/customize-schema.js
@@ -6,6 +6,7 @@ const {
   GraphQLString,
   GraphQLInt,
   GraphQLFloat,
+  GraphQLNonNull,
 } = require(`gatsby/graphql`)
 const {
   queueImageResizing,
@@ -73,10 +74,10 @@ const fixedNodeType = ({
             }),
         },
         aspectRatio: { type: GraphQLFloat },
-        width: { type: GraphQLFloat },
-        height: { type: GraphQLFloat },
-        src: { type: GraphQLString },
-        srcSet: { type: GraphQLString },
+        width: { type: new GraphQLNonNull(GraphQLFloat) },
+        height: { type: GraphQLNonNull(GraphQLFloat) },
+        src: { type: GraphQLNonNull(GraphQLString) },
+        srcSet: { type: GraphQLNonNull(GraphQLString) },
         srcWebp: {
           type: GraphQLString,
           resolve: ({ file, image, fieldArgs }) => {
@@ -229,9 +230,9 @@ const fluidNodeType = ({
               reporter,
             }),
         },
-        aspectRatio: { type: GraphQLFloat },
-        src: { type: GraphQLString },
-        srcSet: { type: GraphQLString },
+        aspectRatio: { type: new GraphQLNonNull(GraphQLFloat) },
+        src: { type: new GraphQLNonNull(GraphQLString) },
+        srcSet: { type: new GraphQLNonNull(GraphQLString) },
         srcWebp: {
           type: GraphQLString,
           resolve: ({ file, image, fieldArgs }) => {
@@ -266,7 +267,7 @@ const fluidNodeType = ({
             ).then(({ srcSet }) => srcSet)
           },
         },
-        sizes: { type: GraphQLString },
+        sizes: { type: new GraphQLNonNull(GraphQLString) },
         originalImg: { type: GraphQLString },
         originalName: { type: GraphQLString },
         presentationWidth: { type: GraphQLInt },

